ripe/atlas/tools/renderers/traceroute_aspath.py

```diff
@@ -0,0 +1,78 @@
+import json
+import urllib2
+
+from ..ipcache import ip_cache
+from .base import Renderer as BaseRenderer
+from .base import Result
+
+
+class Renderer(BaseRenderer):
+
+    RENDERS = [BaseRenderer.TYPE_TRACEROUTE]
+
+    def __init__(self):
+        self.paths = {}
+
+        # Number of different ASs starting from the end of the traceroute path.
+        #
+        #TODO: if a method to pass options to renderers will be defined
+        # this could be set by user input.
+        self.RADIUS = 2
+        
+    @staticmethod
+    def _get_asns_for_output(asns,radius):
+        asns_with_padding = [""]*radius + asns
+        asns_with_padding = asns_with_padding[-radius:]
+        return " ".join(["{:>8}".format("AS{}".format(asn) if asn else "") for asn in asns_with_padding])
+
+    def on_start(self):
+        return "For each traceroute path toward the target, the " \
+               "last {} ASNs will be shown\n\n".format(self.RADIUS)
+        
+    def on_result(self, result, probes=None):
+
+        ip_hops = []
+
+        for hop in result.hops:
+            for packet in hop.packets:
+                if packet.origin:
+                    ip_hops.append(packet.origin)
+                    break
+
+        asns = []
+
+        # starting from the last hop's IP, get up to <RADIUS> ASNs
+        for ip in ip_hops[::-1]:
```

Comment:
See my comments on the IpCache code you've already got, but this might be better represented as

```python
for address in reversed(ip_hops):  # Use of [::-1] is discouraged
    ip = Ip(address)  # The new Ip object from your other class
    ...
```